compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -734,7 +735,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                         if !ptr_valid {
                             throw_ub!(InvalidTag(dbg_val))
                         }
-                        dataful_variant
+                        assert!(!is_relative);
```

Comment:
Why does it make sense to assert `!is_relative` based on the fact that we are looking at an enum that contains a pointer...?

Reply:
It's a sanity check, because `AddNicheCases` should not have applied in this case - we cannot have more than two variants for a niche-optimized pointer.

Reply:
I added a comment to this effect.

Reply:
How is the number of variants relevant?

It doesn't matter what our *current* optimizations happen to do, the question is whether the MIR program at hand has well-defined behavior.

Generally, if anything in Miri mentions any optimizations, something is wrong. Miri implements the MIR semantics. Those are not defined by optimizations, it is the other way around -- optimizations are constrained by the MIR semantics.

Reply:
`AddNicheCases` will not apply if there are only two variants, because LLVM will do a better job of optimizing that case in general.

But makes sense, I will remove this assertion.